test/e2e/targets/awssqs/main.go

```diff
@@ -0,0 +1,295 @@
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	"github.com/triggermesh/triggermesh/test/e2e/framework"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/apps"
+	e2esqs "github.com/triggermesh/triggermesh/test/e2e/framework/aws/sqs"
+	e2ece "github.com/triggermesh/triggermesh/test/e2e/framework/cloudevents"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/ducktypes"
+)
+
+/* This test suite requires:
+
+   - AWS credentials in whichever form (https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#hdr-Sessions_options_from_Shared_Config)
+   - The name of an AWS region exported in the environment as AWS_REGION
+*/
+
+var targetAPIVersion = schema.GroupVersion{
+	Group:   "targets.triggermesh.io",
+	Version: "v1alpha1",
+}
+
+const (
+	targetKind     = "AWSSQSTarget"
+	targetResource = "awssqstargets"
+)
+
+const awsAccessKeyIDSecretKey = "awsApiKey"
+const awsSecretAccessKeySecretKey = "awsApiSecret"
+
+var _ = Describe("AWS SQS target", func() {
+	f := framework.New("awssqstarget")
+
+	var ns string
+
+	var trgtClient dynamic.ResourceInterface
+
+	var queueURL string
+	var queueARN string
+	var awsCreds credentials.Value
+	var awsSecret *corev1.Secret
+
+	BeforeEach(func() {
+		ns = f.UniqueName
+
+		gvr := targetAPIVersion.WithResource(targetResource)
+		trgtClient = f.DynamicClient.Resource(gvr).Namespace(ns)
+	})
+
+	Context("a target watches an existing queue", func() {
+		var trgtURL *url.URL
+		var sqsClient sqsiface.SQSAPI
+
+		BeforeEach(func() {
+			sess := session.Must(session.NewSession())
+			sqsClient = sqs.New(sess)
+			awsCreds = readAWSCredentials(sess)
+
+			By("creating a SQS queue", func() {
+				queueURL = e2esqs.CreateQueue(sqsClient, f)
+			})
+
+			By("creating an AWSSQSTarget object", func() {
+				awsSecret := createAWSCredsSecret(f.KubeClient, ns, awsCreds)
+				queueARN = e2esqs.QueueARN(sqsClient, queueURL)
+
+				trgt, err := createTarget(trgtClient, ns, "test-",
+					withARN(queueARN),
+					withCredentials(awsSecret.Name),
+				)
+				Expect(err).ToNot(HaveOccurred())
+
+				trgt = ducktypes.WaitUntilReady(f.DynamicClient, trgt)
+
+				trgtURL = ducktypes.Address(trgt)
+				Expect(trgtURL).ToNot(BeNil())
+			})
+		})
+
+		AfterEach(func() {
+			By("deleting SQS queue "+queueURL, func() {
+				e2esqs.DeleteQueue(sqsClient, queueURL)
+			})
+		})
+
+		When("an event is sent to the target", func() {
+			var sentEvent *cloudevents.Event
+
+			BeforeEach(func() {
+				By("sending an event", func() {
+					sentEvent = e2ece.NewEvent(f)
+
+					job := e2ece.RunEventSender(f.KubeClient, ns, trgtURL.String(), sentEvent)
+					apps.WaitForCompletion(f.KubeClient, job)
+				})
+			})
+
+			It("forwards the event", func() {
```

Comment:
Instead of "forwards the event", I'd try to be more specific like "puts a message onto the queue", just to make it extra clear what is supposed to happen from the user's perspective.